### PR TITLE
Add gluster and xylem

### PR DESCRIPTION
**Pull request by jerith**
Because container volumes are handy.

### CODE REVIEW

**Comment by jerith** [comment]
This isn't done yet, but it shows the general idea.

**Comment by JayH5** [comment]
It's still not super clear to me how all the xylem/gluster parts fit together but this looks reasonable to me. This is the right level of abstraction for seed_stack I think.

Needs tests obviously.

**Comment by calston** [comment]
:+1:

File: manifests/worker.pp [hunk 1]
```diff
@@ -171,4 +179,12 @@
     ensure => $docker_ensure,
     dns    => $advertise_addr,
   }
+
+  if $xylem_backend {
```
Comment (JayH5) [inline manifests/worker.pp:183]: `undef` is falsey.. but I guess we should be consistent with `if $var {...` vs `if $var != undef`.. not sure if you have a preference?

Comment (jerith) [inline manifests/worker.pp:183]: I think `if $var {...}` is cleaner for this kind of thing. Are we already using `if $var != undef {...}` elsewhere?

**Comment by JayH5** [comment]
:+1: (My question above is just nitpicking.. change/respond if you like)

File: manifests/worker.pp (hunk 1 above)
Comment (JayH5) [inline manifests/worker.pp:183]: I'm pretty sure we do somewhere.. just can't think of where off-hand. Agree that this is cleaner.

**Pull request merged**